ui/pages/confirmations/components/alerts/alert-modal/alert-modal.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { useCallback } from 'react';
+import { ButtonVariant } from '@metamask/snaps-sdk';
+import {
+  Box,
+  Button,
+  ButtonSize,
+  Checkbox,
+  Icon,
+  IconName,
+  IconSize,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+} from '../../../../../components/component-library';
+import {
+  AlignItems,
+  BackgroundColor,
+  BlockSize,
+  BorderRadius,
+  Display,
+  IconColor,
+  Severity,
+  TextAlign,
+  TextColor,
+  TextVariant,
+} from '../../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+import useAlerts from '../../../hooks/useAlerts';
+
+export type AlertModalProps = {
+  /** The unique identifier of the entity that owns the alert */
+  ownerId: string;
+  /** The function to be executed when the button in the alert modal is clicked */
+  handleButtonClick: () => void;
+  /** The unique key representing the specific alert field */
+  alertKey: string;
+  /** The function to be executed when the modal needs to be closed */
+  onClose: () => void;
+  /** Customizable button  */
+  customButton?: {
```

Comment:
For clarity, should we remove this for now and add this dynamic button support in the future ticket we discussed?

It's also likely they'll be defined in the `Alert` itself as `actions` rather than as explicit buttons in these modal properties.

Reply:
Make sense, removed. Let's take that in the next ticket.